lib/diagnostic.js

```diff
@@ -8,22 +8,45 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
+  let result = arguments[0];
 
+  for (let i = 0; i < arguments.length; i++) {
+    if (result > arguments[i]) {
+      result = arguments[i];
+    }
+  }
+  return result;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
+  let result = 0;
 
+  for (let i = 0; i < arguments.length; i++) {
+    result += arguments[i];
+  }
+  return result;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-const newArray = function newArray(/* length, defaultValue */) {
+const newArray = function newArray(length, defaultValue) {
+  let result = [];
 
+  for (let i = 0; i < length; i++) {
+    if (typeof defaultValue === "function") {
+      result[i] = defaultValue();
+    }
+    else {
+      result[i] = defaultValue;
+    }
+  }
+
+  return result;
 };
```

Comment:
Very close! For question 4, note that if `defaultValue` is a function, it is supposed to be invoked "with the element's index as an argument." So what you should have in line 42 is actually `result[i] = defaultValue(i)`

Check out this solution:
```
const newArray = function newArray(length, def) {
  let defFunc = def instanceof Function ? def : () => def;
  let array = new Array(length);
  for (let i = 0; i < length; i++) {
    array[i] = defFunc(i);
  }
  return array;
};
```